chain/chain/src/genesis.rs

```diff
@@ -261,3 +276,66 @@ fn get_genesis_congestion_info(
     tracing::debug!(target: "chain", ?shard_id, ?state_root, ?congestion_info, "Computed genesis congestion info.");
     Ok(congestion_info)
 }
+
+#[cfg(test)]
+mod test {
+    use std::str::FromStr;
+
+    use near_async::time::FakeClock;
+    use near_chain_configs::test_genesis::{TestEpochConfigBuilder, TestGenesisBuilder};
+    use near_epoch_manager::EpochManager;
+    use near_primitives::hash::CryptoHash;
+    use near_primitives::types::Balance;
+    use near_primitives::version::PROD_GENESIS_PROTOCOL_VERSION;
+    use near_store::test_utils::create_test_store;
+    use num_rational::Rational32;
+
+    use crate::test_utils::{KeyValueRuntime, MockEpochManager};
+    use crate::{Chain, ChainGenesis};
+
+    #[test]
+    fn test_prod_genesis_protocol_version_block_consistency() {
```

Comment:
This test ensures that future changes to the code should most likely (not 100% guarantee) not change the genesis initialization and genesis block for mainnet/testnet.

This is probably the only safeguard we have specific to protocol version 29 (hardcoded mainnet genesis)